### PR TITLE
Binary Relevance aggregation

### PR DESCRIPTION
**Pull request by denaxen**
## Description
I have added code for Binary Relevance aggregation - simple method for multi-label classification. This approach treats each label as a class in binary classification task and aggregates it separately.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation and examples improvement (changes affected documentation and/or examples)

## Checklist:

- [x] I have read the **CONTRIBUTING** document.
- [x] I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: crowdkit/aggregation/classification/multi_binary.py [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+        task_to_labels: tp.DefaultDict[tp.Union[str, float], tp.List[tp.Union[str, float]]] = defaultdict(list)
+
+        for i, label in enumerate(mlb.classes_):
+            single_label_df = data[['task', 'worker']]
+            single_label_df['label'] = binarized_labels[:, i]
+
+            label_aggregator = self.aggregator(**self.args)
+            label_aggregator.fit_predict(single_label_df)
+            self.aggregators_[label] = label_aggregator
+            for task, label_value in dict(label_aggregator.labels_).items():
```
Comment (dustalov) [inline crowdkit/aggregation/classification/multi_binary.py:88]: Can we enumerate it more elegant than by building a `dict` and then iterating over `items()`?

File: tests/aggregation/test_multi_binary_aggregation.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+import pandas as pd
+import pytest
+from pandas.testing import assert_series_equal
+import typing as tp
+
+from crowdkit.aggregation import MultiBinary, MajorityVote, DawidSkene, Wawa, GLAD
+
+
+@pytest.fixture
+def data_toy() -> pd.DataFrame:
```
Comment (dustalov) [inline tests/aggregation/test_multi_binary_aggregation.py:13]: We already have a fixture with a very similar name. Please rename it to reflect that it is for multi-label data.

File: tests/aggregation/test_multi_binary_aggregation.py [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+            ['t2', 'w1', ['car']],
+            ['t2', 'w2', ['car', 'human']],
+            ['t2', 'w3', ['train']],
+        ],
+        columns=['task', 'worker', 'label']
+    )
+
+
+@pytest.fixture
+def multi_binary_toy_result() -> pd.Series:
```
Comment (dustalov) [inline tests/aggregation/test_multi_binary_aggregation.py:28]: Same comment as for `data_toy()`.

File: crowdkit/aggregation/classification/multi_binary.py [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+import pandas as pd
+from sklearn.preprocessing import MultiLabelBinarizer
+from . import MajorityVote
+from collections import defaultdict
+
+from ..base import BaseClassificationAggregator
+
+
+@attr.s
+class MultiBinary(BaseClassificationAggregator):
```
[the remark this answers is not in the document]
Comment (denaxen) [reply crowdkit/aggregation/classification/multi_binary.py]: Yes, it's called Binary Relevance - very common name

File: tests/aggregation/test_binary_relevance_aggregation.py [hunk 5]
```diff
@@ -0,0 +1,60 @@
+"""
+Simple aggregation tests.
+"""
+import pandas as pd
+import pytest
+from pandas.testing import assert_series_equal
+import typing as tp
```
Comment (dustalov) [inline tests/aggregation/test_binary_relevance_aggregation.py:7]: Please import the needed symbols directly from `typing`.

File: crowdkit/aggregation/classification/binary_relevance.py [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+        data = data[['task', 'worker', 'label']]
+        mlb = MultiLabelBinarizer()
+        binarized_labels = mlb.fit_transform(data['label'])
+        task_to_labels: Dict[Union[str, float], List[Union[str, float]]] = dict()
+
+        for i, label in enumerate(mlb.classes_):
+            single_label_df = data[['task', 'worker']]
+            single_label_df['label'] = binarized_labels[:, i]
+
+            label_aggregator = self.aggregator(**self.args)
```
Comment (dustalov) [inline crowdkit/aggregation/classification/binary_relevance.py:84]: Note that you are trying to create a new instance of the `aggregator` class. In a similar case, sklearn works with already created instances and does not create them: https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.GridSearchCV.html. Let us keep the API similar to them.

File: tests/aggregation/test_binary_relevance_aggregation.py [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+
+@pytest.fixture
+def binary_relevance_toy_result() -> pd.Series:
+    result = pd.Series([['house', 'tree'], ['car']], index=['t1', 't2'], name='agg_label')
+    result.index.name = 'task'
+    return result
+
+
+@pytest.mark.parametrize(
+    'aggregator, args', [(MajorityVote, {}), (DawidSkene, {'n_iter': 10}), (Wawa, {}), (GLAD, {})]
```
Comment (dustalov) [inline tests/aggregation/test_binary_relevance_aggregation.py:35]: See my comment on sklearn above. Here we pass the classes, but sklearn passes the instances. It might be confusing to those who are used to sklearn.

**Review by dustalov — CHANGES_REQUESTED** [review]

File: crowdkit/aggregation/utils.py [hunk 8]
```diff
@@ -17,6 +18,28 @@
...
+    Args:
+        aggregator (BaseClassificationAggregator): aggregator instance to be cloned
+
+    Returns:
+        BaseClassificationAggregator: cloned aggregator's instance. Its params are same to input,
+            except for the results of previous fit (private attributes).
+    """
+    assert isinstance(aggregator, base.BaseClassificationAggregator), \
+        'Can\'t clone object that is not inherit BaseClassificationAggregator'
+    klass = aggregator.__class__
```
Comment (DrhF) [inline crowdkit/aggregation/utils.py:35]: I suggest `aggregator_class` or something like that instead of `klass`

File: crowdkit/aggregation/classification/binary_relevance.py [hunk 9]
```diff
@@ -0,0 +1,118 @@
...
+            A pandas.Series indexed by `task` such that `labels.loc[task]`
+            is the tasks' aggregated labels.
+
+        aggregators_ (dict[str, BaseClassificationAggregator]): Labels' aggregators matched to classes.
+            A dictionary that matches aggregators to classes.
+            The key is the class found in the source data,
+            and the value is the aggregator used for this class.
+            The set of keys is all the classes that are in the input data.
+    """
+    aggregator: BaseClassificationAggregator = attr.ib(
```
Comment (DrhF) [inline crowdkit/aggregation/classification/binary_relevance.py:61]: Suggest `base_aggregator` instead of `aggregator`. I would also consider making it private property

**Review by DrhF — DISMISSED** [review]

File: crowdkit/aggregation/classification/binary_relevance.py (hunk 9 above)
Comment (denaxen) [reply crowdkit/aggregation/classification/binary_relevance.py]: I agree with renaming, but I don't think it's a good idea to make it private because this is a parameter of the class🤔
